TOOLS/gen-interface-changes.py

```diff
@@ -33,16 +33,15 @@ def add_new_entries(docs_dir, out, git):
             timestamp = check_output([git, "log", "--format=%ct", "-n", "1", "--",
                                       f], encoding="UTF-8")
             if timestamp:
-                files.append((f, timestamp))
+                content = f.read_text()
+                files.append(content)
             else:
                 print(f"Skipping file not tracked by git: {f.name}")
 
-    files.sort(key=lambda x: x[1])
-    for file in files:
-        with open(file[0].resolve(), "r") as f:
-            for line in f:
-               line =  "    - " + line.rstrip()
-               out.write(line + "\n")
+    for file in reversed(sorted(files)):
```

Comment:
Please add a comment explaining why `reversed` "works". Nitpick, is that remove is imho more important than rename, but doesn't matter in practice.

Reply:
Also have you considered grouping the changes by type with separated sections? I think ti would be more readable. Although it would break existing file format.